functionary/ui/views/teams/delete.py

```diff
@@ -0,0 +1,31 @@
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import get_object_or_404
+from django.views.generic import View
+
+from core.auth import Permission
+from core.models import Environment, EnvironmentUserRole, Team, TeamUserRole, User
+
+
+class TeamDeleteMemberView(LoginRequiredMixin, UserPassesTestMixin, View):
+    def delete(self, request: HttpRequest, team_id: str, user_id: str):
+        team = get_object_or_404(Team, id=team_id)
+        user = get_object_or_404(User, id=user_id)
+
+        _ = TeamUserRole.objects.get(user=user, team=team).delete()
+
+        """
+        Delete all inherited relationships for the user
+        within all environments owned by the team.
+        """
+        environments = Environment.objects.filter(team=team)
+        for environment in environments:
+            _ = EnvironmentUserRole.objects.filter(
+                user=user, environment=environment, inherited=True
+            ).delete()
```

Comment:
This shouldn't be tracked.

Reply:
Why would you not want to remove a user from all the team-owned environments if they get removed from the team? I interpreted the team as being the central management component for all the people who have access to the team's environments

Reply:
No, this interpretation is not correct.  You can be on an environment without being a member of the team.  Think of it like being a member of an organization in GitHub versus being given direct access to a repository within the organization.

A user with a role on a Team does not actually have that role on the Environment.  The permissions from the Team role are effectively inherited at the environment level.  So if you ask the question, "does this user have TASK_READ permission for the environment?" the way we answer that is by first checking if they have the permission for the environment.  If not, we check if they have that permission for the team.  So there is no need to explicitly create an environment role for the user when we assign a team role.

Reply:
But if you wanted a user on the team to have a different role within the environment, they would have to have an environment role, right? I create the environment roles to provide that relationship to instantiate their default environment roles based on their team role, and let their environment role change just within the environment

Reply:
Yes to your first statement.  But you don't need to establish a default environment role based on the the team role.  The permission checking behavior just works without that and creating the actual environment role entry does not give you access to any more information than you would have without it.

Reply:
So in my implementation, I treat the team as just the sub-set of accesses to every environment owned by the team, while the environments maintain a super-set of accesses with respect to the team. The permissions a user has in an environment can be entirely separate from the role they have on a team, but by default, they just inherit their role from the team. In order for the team members to have different roles within each environment, I create that EnvironmentRole relationship across all the team environments when the user has been added to the team. I believe this approach is very similar to what Gitlab does for managing users within subgroups, which is what I figured we were going for

Reply:
Maybe this is that part that's tripping you up:  "The permissions a user has in an environment can be entirely separate from the role they have on a team".

That statement is not correct.  The Team role establishes the minimum permissions that a user will have for all of the Environments of that Team.  A user assigned a role at the Team level will always have those permissions for all Environments of that Team.  For example, if a user has the "Admin" role on a Team and then someone explicitly assigns them the "Read Only" role for one of the Environments, the user still has all of the "Admin" permissions.  That Environment role assignment is effectively meaningless, unless the Team role assignment gets removed one day.

Bottom line:  There's no need to introduce this inherited role concept in order to have the permissions work the way they are intended to work today.  If there's a disagreement about how the permissions should work we can have that conversation, but in the context of this PR we'll want to drop the whole inheritance thing.

Reply:
Ok, yea that is the part that I am confused about. I thought we were using the Gitlab model where the Gitlab team establishes the minimum permissions a user has across all repos owned by the subgroup, and then that user can be assigned different roles inside each repo as needed.

So then the EnvironmentUserRole is really just for establishing a relationship between a user and an environment, when the user is not part of the team that owns the environment?